### PR TITLE
Add support for AdvancedVanish

### PR DESCRIPTION
**Pull request by ElBananaa**
See https://github.com/kangarko/ChatControl-Red/issues/2085 for the feature request.

This has been tested, and it appears to be working perfectly fine.

### CODE REVIEW

File: src/main/java/org/mineacademy/fo/model/HookManager.java [hunk 1]
```diff
@@ -1662,6 +1691,18 @@ public static void sendDiscordMessage(final String channel, @NonNull final Strin
...
 
+class AdvancedVanishHook {
+
+	boolean isVanished(final Player player) {
+		return player != null && AdvancedVanishAPI.INSTANCE.isPlayerVanished(player);
+	}
+
+	void setVanished(Player player, boolean vanished) {
+		if (player != null && isVanished(player) != vanished)
+			setVanished(player, false);
```
Comment (kangarko) [inline src/main/java/org/mineacademy/fo/model/HookManager.java:1702]: this method calls itself and causes infinite loop, next time please test

File: src/main/java/org/mineacademy/fo/model/HookManager.java [hunk 2]
```diff
@@ -801,7 +815,19 @@ public static boolean isVanishedCMI(final Player player) {
...
-	 * Sets the vanish status for player in CMI and Essentials
+	 * Return true if the given player is vanished in AdvancedVanish
+	 *
+	 * @deprecated this does not call metadata check for most plugins nor NMS check, see {@link PlayerUtil#isVanished(Player)}
+	 * @param player
+	 * @return
+	 */
+	@Deprecated
+	public static boolean isVanishedAdvancedVanish(final Player player) {
+		return isAdvancedVanishLoaded() && AdvancedVanishAPI.INSTANCE.isPlayerVanished(player);
```
Comment (kangarko) [inline src/main/java/org/mineacademy/fo/model/HookManager.java:826]: you could just have called advancedVanish.isVanished since this calls the same

**Comment by kangarko** [comment]
Awesome thanks! Added two notes there for next time.

File: src/main/java/org/mineacademy/fo/model/HookManager.java (hunk 1 above)
Comment (ElBananaa) [reply src/main/java/org/mineacademy/fo/model/HookManager.java:1702]: > this method calls itself and causes infinite loop, next time please test

This looks weird, I did run tests and didn't experience any issues

**Pull request merged**